mantidproject/mantid
# Elemental Analysis Menu Bar

**Description of work.**
- Add a basic menu bar with the option of changing the default peak data file
- Reduce the size of the Periodic Table

**Report to:** Aidy

**To test:**
- Interfaces -> Muon -> Elemental Analysis
- The table should behave as detailed in #23074.
- Edit -> `Change Peak Data file` should bring up a file browser where a singular file can be selected
- Loading the given JSON (save as a file, then load) should disable all elements other than `Ag`, and clicking `Ag` should print the corresponding peaks
- If elements are selected (yellow in colour) before the peak data file is changed, they should not be selected upon changing the file, although there will be a black box around the most recently selected element
- Loading non-json files should result in a parsing error, and the selection should not have changed
```json
{
  "Ag": {
    "Z": 47,
    "A": 107.87,
    "Primary": {
      "K(4->1)": 3177.7,
      "K(5->1)": 3184.9,
      "L(3->2)": 868.1,
      "L(4->2)": 900.7,
      "M(4->3)": 304.7,
      "6->5": 141
    }
  }
}
```

Fixes #23115.

*This does not require release notes* because they were added in a previous PR.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
